### PR TITLE
Fixes for NVFP4 numerical accuracy for router GEMM output and wrong correction bias cast

### PR DESCRIPTION
**Pull request by b8zhong**
When in quick testing, the accept len go from `3.413 -> 3.894`

When OSL is 2048 using generate,

Before:
len = 3.459, tps = 304.65

After:
accept len = 3.842, tps = 334.12

when add the second commit of this pr,
len = 3.879, tps = 335.94 (the same basicaly)

Firstly, it's wrong to cast correction bias from fp32 to bf16 (confirmed when loading checkpoint values the accuracy of these values change), secondly, the router gemm output needs fp32 output for numerical stability, which is done by trtllm, and the compute dtype is still bf16, thus the performance loss should be mininmal.

---
### CI States

Latest PR Test (Base): :x: [Run #28703167425](https://github.com/sgl-project/sglang/actions/runs/28703167425)
Latest PR Test (Extra): :x: [Run #28703167363](https://github.com/sgl-project/sglang/actions/runs/28703167363)

### CODE REVIEW

File: python/sglang/srt/models/deepseek_v2.py [hunk 1]
```diff
@@ -511,16 +506,13 @@ def forward(
 
             elif _use_aiter:
                 logits = aiter_dsv3_router_gemm(hidden_states, self.weight)
-            elif _is_npu:
+            elif _is_npu or not _is_cuda:
```
Comment (mmangkad) [inline python/sglang/srt/models/deepseek_v2.py]: Nit
```suggestion
            elif not _is_cuda:
```

**Review by mmangkad — APPROVED** [review]
Good catch

**Comment by kpham-sgl** [comment]
## Acceptance-length benchmark (both commits) — #28343 methodology

Tested both commits with the same rigor as #28343, sweeping the full 8-benchmark card suite and measuring mean accepted draft tokens/step.

**Setup:** target `nvidia/Kimi-K2.5-NVFP4` + EAGLE3 draft `lightseekorg/kimi-k2.5-eagle3-mla`, **TP=8 on 8×B200**, `modelopt_fp4` + `flashinfer_trtllm` MoE (the path commit 1 gates on), fp8 KV, `num_steps=3 / topk=1 / num_draft_tokens=4`. Measured via `sglang.bench_serving` (speed-bench loader, greedy `temp=0`, natural EOS capped at 1024 tokens, concurrency 8, seed 1, radix cache flushed before each benchmark). The three points were isolated by checking out the exact commits (they differ only in `MoEGate`): baseline (`53c61bd`) → +commit1 (`fecfc9a`) → +commit1+commit2 (`04edd52`, full PR). Baseline reproduced the #28343 K2.5 "prior to fix" column, and I confirmed the changed path was actually active (`is_flashinfer_trtllm()` → True, `get_name()=="modelopt_fp4"`, so baseline really downcast the correction bias to bf16 while commit 1 keeps fp32).

| Category | Benchmark | N | baseline | +commit1 | +commit2 (full PR) | Δc1 | Δc2 |
| -------- | --------- | - | -------- | -------- | ------------------ | --- | --- |
| Dialogue | MTBench | 80 | 2.47 | 2.48 | 2.47 | +0.01 | +0.00 |
| Chinese | CEval | 212 | 2.36 | 2.38 | 2.33 | +0.02 | −0.03 |
| Math | GSM8K | 500 | 3.02 | 3.03 | 3.02 | +0.01 | +0.00 |
| Code | HumanEval | 164 | 3.07 | 3.05 | 3.04 | −0.02 | −0.03 |
| Math | MATH500 | 500 | 3.05 | 3.06 | 3.01 | +0.01 | −0.04 |
| Math | AIME | 30 | 2.75 | 2.79 | 2.86 | +0.04 | +0.11 |
| Code | LiveCodeBench | 121 | 2.45 | 2.43 | 2.43 | −0.02 | −0.02 |
| Code | SPEED-Bench (coding) | 80 | 2.97 | 2.97 | 2.97 | +0.00 | +0.00 |
| | **Mean (prompt-weighted)** | | **2.877** | **2.883** | **2.859** | **+0.006** | **−0.018** |

All requests succeeded (no failures).

### Read
On **Kimi-K2.5-NVFP4**, both commits are **flat — within run-to-run noise** (per-benchmark spread ±0.05; #28343's own K2.5 prefix-vs-postfix deltas ranged up to ±0.07, incl. negatives). This does **not** reproduce the reported 3.41 → 3.89 jump. That gap is expected: those numbers come from a different regime (single long `OSL 2048` generate stream, acceptance ~3.4–3.9 vs the chat suite's ~2.4–3.1) and the fix plausibly matters more on **K2.6-NVFP4**. I only have K2.5-NVFP4 cached here; happy to rerun the identical sweep on `nvidia/Kimi-K2.6-NVFP4` + `lightseekorg/kimi-k2.6-eagle3-mla` to line up directly with the reported gains.

**Review by Fridge003 — APPROVED** [review]

**Comment by b8zhong** [comment]
It appears https://github.com/sgl-project/sglang/actions/runs/28703167425/job/85271490374?pr=29783 is not related.
